### DISCUSSION TITLE
Show Popover on day click

### DISCUSSION THREAD

**Discussion by kpoxo6op** [gpbl/react-day-picker#2022] [Support]
How do I properly integrate shadcn Popover with form submission in my Day? The day should be selected on click, popover should be displayed as well.

Here is my best attempt, it is ugly and day selection is disabled: [codesandbox](https://codesandbox.io/p/github/kpoxo6op/popover/master?file=%2Fapp%2Fpage.tsx%3A79%2C15)

**Comment by gpbl** [comment]
Hey thanks for proposing this use case - what is the issue in your current implementation?

**Reply by kpoxo6op** [reply]
The main issue is lack of my react knowledge.

I have two handleClick functions, I can only make the popover working when I comment out the  day selection. The code is a blend of copypastas so I guess it's far from elegant.

I'd like to have the single handleClick function, working day select when I click on the day and less ugly code.
